include/CMakeLists.txt

```diff
@@ -1,10 +1,44 @@
 option(INSTALL_TF_PSA_CRYPTO_HEADERS "Install TF PSA Crypto headers." ON)
 
+# The handling of TF_PSA_CRYPTO_CONFIG_NAME assumes that the default
+# configuration file 'include/psa/crypto_config.h' exists in the source tree.
+# That file is copied into the build tree and then modified with 'config.py'
+# according to the configuration name provided via TF_PSA_CRYPTO_CONFIG_NAME.
+set(TF_PSA_CRYPTO_CONFIG_NAME "" CACHE STRING "Name of the configuration to use (see config.py documentation)")
+
+if(TF_PSA_CRYPTO_CONFIG_NAME)
+    if(TF_PSA_CRYPTO_CONFIG_FILE)
+        message(FATAL_ERROR "TF_PSA_CRYPTO_CONFIG_NAME and TF_PSA_CRYPTO_CONFIG_FILE are incompatible.")
+    endif()
+    file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/psa)
+    configure_file(${PROJECT_SOURCE_DIR}/include/psa/crypto_config.h
```

Comment:
If I understand correctly, `crypto_config.h` in the build tree is created when you run `cmake`, and won't be updated if the file changes in the source tree, but will be re-created if you re-run `cmake`?

(I think this behavior is fine, I just want to check my understanding.)

Reply:
It was not the case in the version you read. It is the case now. I've replaced `configure_file` with a simple file copy.
The configuration file in the build tree is updated only when your configure or re-configure the build tree.
Then if you have configured using the TF_PSA_CRYPTO_CONFIG_NAME option, if you modify crypto_config.h in the source tree, this has no impact on the build of the libraries/tests/programs. If you modify crypto_config.h in the build tree, this impacts the build/re-build of the libraries/tests/programs as expected.